### PR TITLE
Change logging level in certain cases

### PR DESCRIPTION
**Pull request by vjon**
Follow-up to PR #2823

Change the logging level used in a number of locations in the code base, to better capture priority of each message.

Primary changes are cases of LOG(INFO) to VLOG(1), with a few occurrences of LOG(INFO) to LOG(WARNING).

### CODE REVIEW

**Comment by mtmcgrew** [comment]
ok to test

**Comment by osqueryer** [comment]
:no_good: The commit 0fa16d0f4c6acd5e4387dc7efef4d4f0c67ce83d (Job results: [945](https://jenkins.osquery.io/job/osqueryPullRequestAudit/945/)) failed the code audit or documentation test.

File: osquery/events/events.cpp [hunk 1]
```diff
@@ -906,8 +906,8 @@ void attachEvents() {
   for (const auto& subscriber : subscribers) {
     auto status = EventFactory::registerEventSubscriber(subscriber.second);
     if (!status.ok()) {
-      LOG(INFO) << "Error registering subscriber: " << subscriber.first << ": "
-                << status.getMessage();
+      VLOG(1) << "Error registering subscriber: " << subscriber.first << ": "
```
Comment (PoppySeedPlehzr) [inline osquery/events/events.cpp]: Any particular reason this should stay as an info level log versus something like `LOG(WARNING)`? I think I like the change to `VLOG(1)` also, but it seems to me more consistent with the flow of events if this was a `LOG(WARNING)` as above.

Comment (vjon) [inline osquery/events/events.cpp]: My original intention was for the error preventing the loop from happening to have greater prominence than errors for individual subscribers, hence `LOG(WARNING)` above and `VLOG(1)` here. This would also avoid spaming the Error Log if multiple subscribers fail to register.

I also agree that having both be LOG(WARNING) for consistency is another valid approach, and I guess it comes down to what aspects should be prioritized.

**Comment by vjon** [comment]
@PoppySeedPlehzr I too am interested in opinions on whether these verbosity changes are too drastic.

Any and all input from osquery devs/users/etc is welcomed.

File: osquery/core/init.cpp [hunk 2]
```diff
@@ -603,7 +603,7 @@ void Initializer::start() const {
     if (tool_ == ToolType::DAEMON) {
       LOG(WARNING) << message;
     } else {
-      LOG(INFO) << message;
```
Comment (theopolis) [inline osquery/core/init.cpp]: `INFO` is silenced on the shell by default, what is the thought here, why the change?

File: osquery/core/system.cpp [hunk 3]
```diff
@@ -226,7 +226,7 @@ Status checkStalePid(const std::string& content) {
     return Status(1, "osqueryd (" + content + ") is already running");
   } else {
     VLOG(1) << "Found stale process for osqueryd (" << content
-            << ") removing pidfile";
+            << "); Removing pidfile";
```
Comment (theopolis) [inline osquery/core/system.cpp]: ?

File: osquery/database/plugins/rocksdb.cpp [hunk 4]
```diff
@@ -135,7 +135,7 @@ void GlogRocksDBLogger::Logv(const char* format, va_list ap) {
     // logger from trying to make a call back into rocksdb and causing a
     // deadlock
     LoggerForwardingDisabler forwarding_disabler;
-    LOG(INFO) << "RocksDB: " << log_line;
+    VLOG(1) << "RocksDB: " << log_line;
```
Comment (theopolis) [inline osquery/database/plugins/rocksdb.cpp]: Please don't change this. :) This is useful information we want to pick up without turning on verbose.

File: osquery/events/events.cpp [hunk 5]
```diff
@@ -687,8 +687,8 @@ Status EventFactory::registerEventPublisher(const PluginRef& pub) {
     specialized_pub->state(EventState::EVENT_SETUP);
     if (!status.ok()) {
       // Only start event loop if setUp succeeds.
-      LOG(INFO) << "Event publisher failed setup: " << type_id << ": "
-                << status.what();
+      LOG(WARNING) << "Event publisher failed setup: " << type_id << ": "
```
Comment (theopolis) [inline osquery/events/events.cpp]: These are usually innocuous and already glen too much attention for newcomers, I'd suggest keeping them `INFO`.

File: osquery/core/system.cpp (hunk 3 above)
Comment (vjon) [inline osquery/core/system.cpp]: Just a little nitpicking, as the previous structure resulted in a bit of a run-on sentence (the action "Removing pidfile" is a consequence of the first part of the statement, and should be separated by something like a semicolon or a comma).

File: osquery/core/init.cpp (hunk 2 above)
Comment (vjon) [inline osquery/core/init.cpp]: I was thinking that it would be nice to have the option not to log this (i.e. when verbose logging is turned off), instead of the `LOG(INFO)` default behavior of always logging to `/tmp/<program name>...`

In particular, if the volume of logging throughout osquery continues to increase in the future, it would be useful to have as much "non-essential" stuff migrated to verbose logging such as to not clog up the log files and prevent any (marginal) performance issues.

Of course, a single logging statement probably doesn't have much effect in the long run, and I can revert this change if necessary.

Comment (theopolis) [inline osquery/core/init.cpp]: Ah, I think this behavior is a recently-fixed regression. See: https://github.com/facebook/osquery/commit/976db066c05ee9cdf650c056b641ba5f588185ee

File: osquery/core/system.cpp (hunk 3 above)
Comment (theopolis) [inline osquery/core/system.cpp]: Got it, do you mind just removing "remove pidfile"?

File: osquery/events/events.cpp (hunk 1 above)
Comment (theopolis) [inline osquery/events/events.cpp]: 👍

**Comment by theopolis** [comment]
Few final observations, almost there!

**Comment by osqueryer** [comment]
:-1: The commit ba3fe4789a166e17839d0bc8602c9a98c7838c52 (Job results: [1029](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1029/)) failed one or more tests (**Windows**).

**Comment by theopolis** [comment]
Interesting edge case for the `test_reset` pointer comparison. I'm guessing this is an unhandled condition in the test, where it is possible the pointer values are the same-- though unlikely. We should enhance that test a bit.

**Comment by theopolis** [comment]
ok to test
